src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -193,6 +193,10 @@
     BooleanLiteralGrammar=OneOf(
         Ref("TrueSegment"), Ref("FalseSegment"), Ref("UnknownSegment")
     ),
+    PostFunctionGrammar=Sequence(
```

Comment:
I was tempted to just add `FILTER` as an unreserved keyword to the exasol dialect to overcome this, but exasol doesn't support the filter syntax so I decided to replace the `PostFunctionGrammar` even though this feels verbose.

Reply:
This seems reasonable to me -- not that many lines of additional code. If you wanted to avoid it, one possibility would be to refactor the ANSI `PostFunctionGrammar` into something like:

```
    PostFunctionGrammar=OneOf(
        # Optional OVER suffix for window functions.
        # This is supported in biquery & postgres (and its derivatives)
        # and so is included here for now.
        Sequence(
            Sequence(OneOf("IGNORE", "RESPECT"), "NULLS", optional=True),
            Ref("OverClauseSegment"),
        ),
        Ref("BaseFilterClauseGrammar"),
    ),
```

Then define `BaseFilterClauseGrammar` in ANSI and override it to be blank in EXASOL. I did something slightly similar in #1011.

I'm curious, what was the error you were getting in EXASOL? I encountered an interesting one a week or so ago with EXASOL. I was enhancing the lexer for floating-point values, and I was initially allowing trailing decimals (e.g. `5.`). That caused an error with EXASOL, which supports some sort of range syntax, e.g. (`5..10`). EXASOL has 🐉 🐉 ! 😛